geomstats/learning/frechet_mean.py

```diff
@@ -355,3 +366,87 @@ def fit(self, X, y=None, weights=None):
         self.estimate_ = mean
 
         return self
+
+    def transform(self, X, y=None, base_point=None):
+        """Lift data to a tangent space.
+
+        Compute the logs of all data point and reshapes them to
+        1d vectors if necessary. By default the logs are taken at the mean
+        but any other base point can be passed. Any machine learning
+        algorithm can then be used with the output array.
+
+        Parameters
+        ----------
+        X : array-like, shape=[n_samples, {dim, [n, n]}]
+            Data to transform.
+        y : Ignored (Compliance with scikit-learn interface)
+        base_point : array-like, shape={dim, [n,n]}, optional (mean)
+            Point on the manifold, the returned samples will be tangent
+            vectors at the base point.
+
+        Returns
+        -------
+        X_new : array-like, shape=[n_samples, dim]
+        """
+        if base_point is None:
+            base_point = self.estimate_
+
+            if self.estimate_ is None:
+                raise RuntimeError('fit needs to be called first or a '
+                                   'base_point passed.')
+
+        tangent_vecs = self.metric.log(X, base_point=base_point)
+
+        if self.point_type == 'vector':
+            return tangent_vecs
+
+        if gs.all(Matrices.is_symmetric(tangent_vecs)):
+            X = SymmetricMatrices.vector_from_symmetric_matrix(
+                tangent_vecs)
+        elif gs.all(Matrices.is_skew_symmetric(tangent_vecs)):
```

Comment:
elif and else are not tested.

Reply:
I knoooow, but FrechetMean doesn't work with point_type='matrix' for SOn.bi_invariant_metric (maybe it's fixed with your vectorization...)

Reply:
how about `default_gradient_descent`, I thought this one was working with matrices?

Reply:
yes if metric.log works, but last time I tried with SOn it didn't

Reply:
Got it :/